Source/WebKit/UIProcess/API/APIFeatureStatus.h

```diff
@@ -45,6 +48,28 @@ enum class FeatureStatus : uint8_t {
     Mature
 };
 
+// Helper for representing feature status as a constant type. Used by the preferences generator to
+// validate feature configuration at compile time.
+template<API::FeatureStatus Status>
+class FeatureConstant : public std::integral_constant<API::FeatureStatus, Status> { };
+
+constexpr std::optional<bool> defaultValueForFeatureStatus(FeatureStatus status)
+{
+    switch (status) {
+    case FeatureStatus::Stable:
+    case FeatureStatus::Mature:
+        return true;
+    case FeatureStatus::Unstable:
+    case FeatureStatus::Developer:
+    case FeatureStatus::Testable:
+    case FeatureStatus::Preview:
+        return false;
+    case FeatureStatus::Embedder:
+    case FeatureStatus::Internal:
+        return { };
```

Comment:
Added a comment. It's for two different reasons: Embedder features are intended to vary widely between platforms, so they don't have an implied default value.

Internal features are supposed to be off by default (as documented in `UnifiedWebPreferences.yaml`), but we have a number of existing internal features that break the rule. Since these are not user-visible, I don't think we need to correct them all now—we gain most of the benefits by having internal checking of the public statuses.